src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -51,16 +64,56 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition7(){
-        return null;
+        Map<String, Long> result = new HashMap<>();
+        Multiset<String> roadSurfaceCondition = HashMultiset.create();
+        for (RoadAccident rec : roadAccidentList ) {
+            roadSurfaceCondition.add(rec.getRoadSurfaceConditions());
+        }
+        for (String key : roadSurfaceCondition.elementSet() ) {
+//            System.out.println(key+" count："+roadSurfaceCondition.count(key));
+            result.put(key, Long.valueOf(roadSurfaceCondition.count(key)));
+        }
+        return result;
     }
 
     /**
      * find the weather conditions which caused the top 3 number of incidents
      * as example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1 in foggy, then your result list should contain {rain, sunny, snow} - top three in decreasing order
      * @return
      */
+    public List<String> getTopThreeWeatherCondition6(){
+
+        Multiset<String> weatherConditions = HashMultiset.create();
+        for (RoadAccident rec : roadAccidentList ) {
+            weatherConditions.add(rec.getWeatherConditions());
+        }
+        ImmutableSortedMap.Builder<String, String> builder = new ImmutableSortedMap
+                .Builder<String, String>(Ordering.natural().reverse());
+        for (String key : weatherConditions.elementSet() ) {
+            String countStr = String.format("%08d %s", weatherConditions.count(key), key);
+            System.out.println(countStr);
+            builder.put(countStr, key);
+        }
+        ImmutableSortedMap<String, String> sortedConditions = builder.build();
+        List<String> result = sortedConditions.values().asList();
+//        System.out.println(result);
+        return result.subList(0,3);
```

Comment:
That's a pretty cool solution as well! 
One minor point - subList will throw exception if list has less then 3 elements.